nifi-nar-bundles/nifi-oraclecdc-bundle/nifi-oraclecdc-processors/src/main/java/org/apache/nifi/processors/oraclecdc/controller/impl/StandardOracleCDCService.java

```diff
@@ -0,0 +1,269 @@
+        props.add(DB_HOST);
+        props.add(DB_PORT);
+        props.add(DB_USER);
+        props.add(DB_PASS);
+        props.add(DB_SID);
+        props.add(DB_CLASS_LOADER);
+        props.add(MAX_TOTAL_CONNECTIONS);
+        props.add(MAX_IDLE);
+        props.add(MIN_IDLE);
+
+        properties = Collections.unmodifiableList(props);
+    }
+
+    private volatile BasicDataSource dataSource;
+    // private volatile String dbUrl;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder().name(propertyDescriptorName).required(false)
+                .addValidator(StandardValidators
+                        .createAttributeExpressionLanguageValidator(AttributeExpression.ResultType.STRING, true))
+                .addValidator(StandardValidators.ATTRIBUTE_KEY_PROPERTY_NAME_VALIDATOR)
+                .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY).dynamic(true).build();
+    }
+
+    /**
+     * @param context
+     *            the configuration context
+     * @throws InitializationException
+     *             if unable to create a database connection
+     */
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws InitializationException {
+        final String host = context.getProperty(DB_HOST).getValue();
+        final String port = context.getProperty(DB_PORT).getValue();
+        final String user = context.getProperty(DB_USER).getValue();
+        final String passw = context.getProperty(DB_PASS).getValue();
+        final String dbSid = context.getProperty(DB_SID).getValue();
+        final OracleClassLoaderService clService = context.getProperty(DB_CLASS_LOADER)
+                .asControllerService(OracleClassLoaderService.class);
+        final String drv = "oracle.jdbc.OracleDriver";
+        final Integer maxTotal = context.getProperty(MAX_TOTAL_CONNECTIONS).asInteger();
+        final Integer maxIdle = context.getProperty(MAX_IDLE).asInteger();
+        final Integer miIdle = context.getProperty(MIN_IDLE).asInteger();
+        dataSource = new BasicDataSource();
+        dataSource.setAccessToUnderlyingConnectionAllowed(true);
+        dataSource.setDriverClassName(drv);
+
+        // Optional driver URL, when exist, this URL will be used to locate
+        // driver jar file location
+        dataSource.setDriverClassLoader(clService.getClassLoader());
+        this.driverClassLoader = clService.getClassLoader();
+        final String dburl = "jdbc:oracle:oci:@" + host + ":" + port + ":" + dbSid;
+        // dataSource.setMaxWaitMillis(maxWaitMillis);
+        dataSource.setMaxTotal(maxTotal);
+        dataSource.setMinIdle(miIdle);
+        dataSource.setMaxIdle(maxTotal);
```

Comment:
Should be `maxIdle` instead of `maxTotal`.

Reply:
yup, goog catch